src/ehooks.c

```diff
@@ -198,15 +241,12 @@ ehooks_default_merge_impl(void *addr_a, void *addr_b) {
 static bool
 ehooks_default_merge(extent_hooks_t *extent_hooks, void *addr_a, size_t size_a,
     void *addr_b, size_t size_b, bool committed, unsigned arena_ind) {
-	if (!maps_coalesce) {
```

Comment:
The `maps_coalesce` was a static check; removing it means the non-Windows cases will do the two `iealloc` unconditionally. Maybe preserve the check for now?

Reply:
The reason I figured I wasn't so concerned was that the lookups only ever happen if ehooks_merge_default is invoked directly. But by default, the ehooks_merge call will short-circuit into the `_impl` variant, avoiding the lookup. (So, the only way this function gets invoked is if the user sneaks out one of our internal function pointers and wraps it or something). What do you think about a middle ground where ehooks_merge checks if the merge hook specifically is non-default?

Reply:
Sounds good!